src/Sarif.Viewer.VisualStudio/Services/ILoadSarifLogService2.cs

```diff
@@ -5,6 +5,9 @@
 
 namespace Microsoft.Sarif.Viewer.Services
 {
+    /// <summary>
+    /// Interface for loading SARIF logs from multiple files into the viewer.
+    /// </summary>
     internal interface ILoadSarifLogService2
```

Comment:
>ILoadSarifLogService2 [](start = 23, length = 21)

I don't understand the strategy of evolving these interfaces. For one thing, we own most of this architecture outright, why are we being so precious about maintenance? I'm not aware of a significant customer of these interfaces. Second, in the COM world a '2' interface would typically inherit from and extend the original version.

Can we simply converge these into a single interface? Yes, I know this is a very COM unfriendly thing to do, back in 1996 it certainly wouldn't fly. Today (in this specific context), I'm not sure I care.

Reply:
Gabe and I discussed both points, but happy to reopen the discussion.

1. Re: numbered interfaces extending earlier interfaces: VS is a mixed bag here. For example, `IVsWindowFrameEvents2` extends `IVsWindowFrameEvents`, but `IVsWindowFrameNotify2` does not extend `IVsWindowFrameNotify`. I have no objection to adding the extensions, subject to point #2.

2. Re: COM unfriendliness, two considerations:
- As old as COM is, VS still uses it under the covers.
- Regardless of COM, adding a method to a "published" (more on that below) interface is a breaking change.

Then the question becomes, are there any customers of these interfaces. The only customer I'm aware of is WAVE -- but! they access the interfaces through the Viewer Interop API -- which uses `dynamic`, and so is _not_ a compile time customer of the interface.

It comes down to whether we think anyone has every taken a build-time dependency on the Viewer's DLL (which is where those interfaces are defined), and then defined a type that implements `ILoadSarifLogService`.



---
In reply to: [520680157](https://github.com/microsoft/sarif-visualstudio-extension/pull/259#discussion_r520680157) [](ancestors = 520680157)

Reply:
The point is not VS general use of COM, the point is whether its use of COM requires us to maintain the level of care around COM interfaces that are required in some scenarios.

My guidance is to simply converge everything into ILoadSarifLogService. I'm not aware of any utilization we would break. If we do break someone, I'd be glad to help fix whatever is required.

---
In reply to: [520702295](https://github.com/microsoft/sarif-visualstudio-extension/pull/259#discussion_r520702295) [](ancestors = 520702295,520680157)

Reply:
Ok, I will merge the interfaces as part of my current PR (which BTW is around displaying the info bar if we receive invalid JSON).

---
In reply to: [520725161](https://github.com/microsoft/sarif-visualstudio-extension/pull/259#discussion_r520725161) [](ancestors = 520725161,520702295,520680157)

Reply:
Please fix the interfaces here, so this change is coherent. i.e., i'd prefer that we not introduce a 3rd interface here then remove it in an arbitrary future PR. The adjustment is small. Let me know when it's in and I'll merge this otherwise approved PR.

---
In reply to: [520726644](https://github.com/microsoft/sarif-visualstudio-extension/pull/259#discussion_r520726644) [](ancestors = 520726644,520725161,520702295,520680157)

Reply:
As discussed offline: This change is already merged, and we agree it's not worth the effort to have a separate "merge interfaces" PR. We will add this to the next PR.

---
In reply to: [520729099](https://github.com/microsoft/sarif-visualstudio-extension/pull/259#discussion_r520729099) [](ancestors = 520729099,520726644,520725161,520702295,520680157)